src/components/Table/index.tsx

```diff
@@ -1,42 +1,88 @@
   stripes,
   columns,
-  data
+  data,
+  name,
+  noDataPlaceholder = (
+    <div className="text-center">
+      <InformationCircleIcon className="mx-auto h-12 text-gray-400" />
+      <h3 className="mt-2 text-sm font-medium text-gray-500">
+        No {name || 'data'}
+      </h3>
+      <p className="mt-1 text-sm text-gray-500">
+        There {name ? 'are' : 'is'} currently no {name || 'data'} to display
+      </p>
+    </div>
+  ),
+  pagination,
 }) => (
   <div className="flex flex-col">
     <div className="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8">
       <div className="py-2 align-middle inline-block min-w-full sm:px-6 lg:px-8">
-        <div className="shadow overflow-hidden border-b border-gray-200 sm:rounded-lg">
+        <div
+          className={`${
+            !pagination && 'border-b rounded-b-lg'
+          } rounded-t-lg shadow overflow-hidden border-gray-200`}
+        >
           <table className="min-w-full divide-y divide-gray-200">
             <thead className="bg-gray-50">
-            <tr>
-              {columns.map(column => (
-                <th
-                  scope="col"
-                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
-                >
-                  {column}
-                </th>
-              ))}
-            </tr>
-            </thead>
-            <tbody>
-            {data.map((row, i) => (
-              <tr key={row.id} className={!stripes || i % 2 === 0 ? 'bg-white' : 'bg-gray-50'}>
-                {columns.map(column => (
-                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">{row[column]}</td>
+              <tr>
+                {columns.map((column) => (
+                  <th
+                    scope="col"
+                    className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
+                  >
+                    {column}
+                  </th>
                 ))}
               </tr>
-            ))}
+            </thead>
+            <tbody className={stripes ? '' : 'divide-y divide-gray-200'}>
+              {data.length ? (
+                data.map((row, i) => (
+                  <tr
+                    key={row.id}
+                    className={
+                      !stripes || i % 2 === 0 ? 'bg-white' : 'bg-gray-50'
+                    }
+                  >
+                    {columns.map((column) => (
+                      <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
+                        {row[column]}
+                      </td>
+                    ))}
+                  </tr>
+                ))
+              ) : (
+                <tr>
+                  <td className="bg-white p-6" colSpan={columns.length}>
+                    {noDataPlaceholder}
+                  </td>
+                </tr>
+              )}
             </tbody>
+            {pagination && (
```

Comment:
I'd expect pagination to be a required prop, there's no table in the world that doesn't paginate

Reply:
What about students in a cohort? All students in it are fetched when the cohort is fetched in cohort detail.

Reply:
there could be more students than 1 page tho